tasks/main.yml

```diff
@@ -77,4 +77,4 @@
       include_role:
         name: "{{ role_path }}/roles/rsyslog"
 
-  when: logging_collector | d('rsyslog') == 'rsyslog'
+  when: logging_provider | d(default_logging_provider) == 'rsyslog'
```

Comment:
why the `d(...)` ? `logging_provider` is set in defaults, so it should be always defined.

Reply:
> why the `d(...)` ? `logging_provider` is set in defaults, so it should be always defined.

This is the point.  Since we do include_role above at the line 78, logging_provider is _not_ found in the included role.

```
RUNNING HANDLER [/home/nhosoi/linux-system-roles/logging/tests/roles/linux-system-roles.logging/roles/rsyslog : restart rsyslogd] ***
task path: /home/nhosoi/linux-system-roles/logging/roles/rsyslog/handlers/main.yml:2
fatal: [/home/nhosoi/Downloads/CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2]: FAILED! => {
    "msg": "The conditional check 'logging_provider == 'rsyslog'' failed. The error was: error while evaluating conditional (logging_provider == 'rsyslog'): 'logging_provider' is undefined\n\nThe error appears to be in '/home/nhosoi/linux-system-roles/logging/roles/rsyslog/handlers/main.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: restart rsyslogd\n  ^ here\n"
}
META: ran handlers
```
And please note that I tried to convert the include_role with include_tasks, but it did not work well. (See also https://github.com/linux-system-roles/logging/pull/110#discussion_r419182270)

Reply:
Well, you should have said that the problem is in a handler, this makes a bit more sense. Still quite surprising though because the handler does not actually use `logging_provider`. Let me experiment a bit.

Reply:
> Well, you should have said that the problem is in a handler, ...

Sorry, @pcahyna. I tried including the var in the handler, but it did not work, either... (;_;)

Reply:
Try this workaround:
```yaml
  - name: Include Rsyslog role
    vars:
      __rsyslog_enabled: "{{ logging_enabled }}"
      __rsyslog_purge_original_conf: "{{ logging_purge_confs }}"
      __rsyslog_system_log_dir: "{{ logging_system_log_dir }}"
    include_role:
      name: "{{ role_path }}/roles/rsyslog"
    when: logging_provider == 'rsyslog'
```
I.e. move the `include_role` out of the `block`. For some reason it is the `block` statement that causes the problem if enclosing the `include_role` statement.

Reply:
Thank you, @pcahyna.  I've updated the PR.